src/modules/flow/http-client/http-client.c

```diff
@@ -0,0 +1,600 @@
+    if (!sol_http_param_add(&params,
+        SOL_HTTP_REQUEST_PARAM_HEADER("Accept", "application/json"))) {
+        SOL_WRN("Failed to set query params");
+        sol_http_param_free(&params);
+        return -ENOMEM;
+    }
+
+    type = (const struct http_client_node_type *)sol_flow_node_get_type(node);
+    connection = sol_http_client_request(SOL_HTTP_METHOD_GET, mdata->url,
+        &params, type->request_finished_cb, node);
+
+    sol_http_param_free(&params);
+
+    SOL_NULL_CHECK(connection, -ENOTCONN);
+
+    r = sol_ptr_vector_append(&mdata->pending_conns, connection);
+    if (r < 0) {
+        SOL_WRN("Failed to keep pending connection.");
+        sol_http_client_connection_cancel(connection);
+        return -ENOMEM;
+    }
+
+    return 0;
+}
+
+static int
+check_response(struct http_data *mdata, struct sol_flow_node *node,
+    const struct sol_http_client_connection *connection,
+    struct sol_http_response *response)
+{
+    if (sol_ptr_vector_remove(&mdata->pending_conns, connection) < 0)
+        SOL_WRN("Failed to find pending connection %p", connection);
+
+    if (!response) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Error while reaching %s", mdata->url);
+        return -EINVAL;
+    }
+    SOL_HTTP_RESPONSE_CHECK_API(response, -EINVAL);
+
+    if (!response->content.used) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Empty response from %s", mdata->url);
+        return -EINVAL;
+    }
+
+    if (response->response_code != SOL_HTTP_STATUS_OK) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "%s returned an unknown response code: %d",
+            mdata->url, response->response_code);
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+static int
+check_key(const char *url, struct sol_json_token *token)
+{
+    char *ptr;
+
+    if (strncmp(url, "http://", sizeof("http://") - 1) == 0)
+        ptr = strstr(url + sizeof("http://") - 1, "/");
+    else
+        ptr = strstr(url, "/");
+    SOL_NULL_CHECK(ptr, -EINVAL);
+
+    return !sol_json_token_str_eq(token, ptr, strlen(ptr));
+}
+
+static void
+boolean_request_finished(void *data,
+    const struct sol_http_client_connection *connection,
+    struct sol_http_response *response)
+{
+    struct sol_flow_node *node = data;
+    struct http_data *mdata = sol_flow_node_get_private_data(node);
+    bool result = true;
+
+    if (check_response(mdata, node, connection, response) < 0)
+        return;
+
+    if (streq(response->content_type, "application/json")) {
+        struct sol_json_scanner scanner;
+        struct sol_json_token token, key, value;
+        enum sol_json_loop_reason reason;
+
+        sol_json_scanner_init(&scanner, response->content.data, response->content.used);
+        SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+            if (check_key(mdata->url, &key))
+                continue;
+            if (sol_json_token_get_type(&value) == SOL_JSON_TYPE_TRUE)
```

Comment:
up to this line this should be a common function. then here it would call a `node_type->process_token(node, &value)`. That would contain the block that converts the value to actual type and send a packet on output port.